src/panner.js

```diff
@@ -1,38 +1,85 @@
+import Effect from './effect.js';
+
 import p5sound from './main';
 var ac = p5sound.audiocontext;
 var panner;
 // Stereo panner
 // if there is a stereo panner node use it
 if (typeof ac.createStereoPanner !== 'undefined') {
-  class Panner {
-    constructor(input, output) {
-      this.stereoPanner = this.input = ac.createStereoPanner();
-      input.connect(this.stereoPanner);
-      this.stereoPanner.connect(output);
+  /**
+   * The Panner class allows you to control the stereo
+   * panning of a sound source. It uses the [StereoPannerNode](https://developer.mozilla.org/en-US/docs/Web/API/StereoPannerNode),
+   * which allows you to adjust the balance between the left and right channels of a sound source.
+   *
+   * This class extends <a href = "/reference/#/p5.Effect">p5.Effect</a>.
+   * Methods <a href = "/reference/#/p5.Effect/amp">amp()</a>, <a href = "/reference/#/p5.Effect/chain">chain()</a>,
+   * <a href = "/reference/#/p5.Effect/drywet">drywet()</a>, <a href = "/reference/#/p5.Effect/connect">connect()</a>, and
+   * <a href = "/reference/#/p5.Effect/disconnect">disconnect()</a> are available.
+   *
+   * @class p5.Panner
+   * @extends p5.Effect
+   */
+  class Panner extends Effect {
+    constructor() {
+      super();
+      this.stereoPanner = this.ac.createStereoPanner();
+
+      this.input.connect(this.stereoPanner);
+      this.stereoPanner.connect(this.wet);
     }
 
+    /**
+     * Set the stereo pan position, a value of -1 means the sound will be fully panned
+     * to the left, a value of 0 means the sound will be centered, and a value of 1 means
+     * the sound will be fully panned to the right.
+     * @method pan
+     * @for p5.Panner
+     * @param {Number} value  A value between -1 and 1 that sets the pan position.
+     *
+     * @param {Number} [time] time in seconds that it will take for the panning to change to the specified value.
+     */
     pan(val, tFromNow) {
-      var time = tFromNow || 0;
-      var t = ac.currentTime + time;
-
-      this.stereoPanner.pan.linearRampToValueAtTime(val, t);
+      if (typeof val === 'number') {
+        let time = tFromNow || 0;
+        this.stereoPanner.pan.linearRampToValueAtTime(
+          val,
+          this.ac.currentTime + 0.02 + time
```

Comment:
What does the 0.02 here do?

Reply:
I left that by accident, I don't think it's necessary, thanks for noticing!